### PR TITLE
Find tests recursively

### PR DESCRIPTION
**Pull request by r0man**
Hello @kirbyfan64,

I'm just starting to explore Hy and was looking for a unit test framework and found yours. However I have the need to run tests in nested directories. This patch changes the find-tests function to find those tests as well. I changed the signature of find-tests so it is a bit more flexible, but mainly for testing purpose.

Would you like to merge this?

Thanks, Roman.

### CODE REVIEW

File: hytest.hy [hunk 1]
```diff
@@ -326,21 +326,15 @@
...
       (if (and (f.startswith "test") (= (get (path.splitext f) 1) ".hy"))
         (test-paths.append (path.join root f)))))
   test-paths)
 
 (defn load-tests []
-  (for [p (find-tests)]
+  (for [p (find-tests (getcwd))]
     (mods.append
-      (import-file-to-module (get (path.splitext (path.basename p)) 0) p))))
+     (import-file-to-module (get (path.splitext (path.basename p)) 0) p))))
```
Comment (kirbyfan64) [inline hytest.hy]: I don't think the removal of the space here was intentional... ;)

File: test_hytest.hy [hunk 2]
```diff
@@ -87,3 +90,22 @@
 
 (test-set test-globals
   (test = x 1))
+
+(defn touch [&rest parts]
+  "Join the path `parts` and touch the resulting filename."
+  (.close (open (apply path.join (list parts)) "a")))
```
Comment (kirbyfan64) [inline test_hytest.hy]: Does this not work without the `(list parts)`? I thought it was supposed to.

**Comment by kirbyfan64** [comment]
Thanks! I pointed out two *really* minor things, though, if you're unable fix them, I'll just merge this as-is and fix them myself.

File: test_hytest.hy (hunk 2 above)
Comment (r0man) [inline test_hytest.hy]: Yes, you are right, it does work without the call to list. I removed it.

File: hytest.hy (hunk 1 above)
Comment (r0man) [inline hytest.hy]: Oops, yes, that was my Emacs setup doing auto indentation. Fixed.

**Comment by r0man** [comment]
@kirbyfan64 I addressed your comments. The build hangs since 20 mins on Travis. I'm out now but will take a look tomorrow if you have more suggestions.

**Comment by r0man** [comment]
Ok, the tests just finished.

**Comment by kirbyfan64** [comment]
@r0man Thanks again! I've got to take a look at speeding up the Travis runs...

**Pull request merged**